torch/csrc/jit/passes/to_batch.cpp

```diff
@@ -0,0 +1,80 @@
+#include "torch/csrc/jit/passes/to_batch.h"
+#include "torch/csrc/jit/script/compiler.h"
+
+namespace torch { namespace jit {
+
+// map from batchTensor to {data, mask, dims}
+static std::unordered_map<Value*, std::vector<Value*>> batch_map;
+
+static void ToBatch(Block* block) {
+  for (auto it = block->nodes().begin(); it != block->nodes().end(); it++) {
+    auto n = *it;
+    // std::cout << *n << std::endl;
+    // replace tensor operator to BatchTensor operator
+    if(n->kind().is_aten()){
+      auto des_graph = batch_operator_table.at(n->kind().toUnqualString());
+      // std::cout << "des_graph:" << *des_graph << std::endl;
+      WithInsertPoint guard(n);
+      std::vector<Value*> new_inputs;
+      for(auto input : n->inputs()){
+        if(batch_map.find(input) != batch_map.end()){
+          auto new_input = batch_map.at(input);
+          new_inputs.insert(new_inputs.end(), new_input.begin(), new_input.end());
+        }
+        else{
+          new_inputs.push_back(input);
+        }
+      }
+      auto outputs = script::inlineCallTo(*n->owningGraph(), *des_graph, new_inputs);
+      for(size_t i = 0; i < n->outputs().size(); i++){
+        auto output = n->outputs()[i];
+        output->replaceAllUsesWith(outputs[i * 3]);
+        batch_map[outputs[i * 3]] = std::vector<Value*>(outputs.begin() + i * 3, outputs.begin() + i * 3 + 3);
+      }
+      it.destroyCurrent();
+    }
+    // control flow
+    else if(n->kind().is_prim()){
+      if(n->kind() == prim::Loop){
+        // TODO
```

Comment:
Can you leave a comment here about why this isn't supported and why we need a special case for prim::Loop?